src/kbc_modules/eaa_kbc/mod.rs

```diff
@@ -149,12 +174,85 @@ impl EAAKbc {
+                "DecryptionResponse status is OK but the data is null"
             ));
         }
     }
 
-    fn kbs_trans_and_recv(&mut self, trans_data: &[u8], error_info: &str) -> Result<String> {
+    fn kbs_get_resource(&mut self, description: String) -> Result<Vec<u8>> {
+        let desc: ResourceDescription =
+            serde_json::from_str::<ResourceDescription>(description.as_str())?;
+
+        let command = format!("Get {}", desc.name.as_str());
+        let request = GetResourceRequest::new(&command, desc.optional.clone());
+        let resource_info = self.kbs_get_resource_info(desc.name.as_str())?;
+
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+        let buffer_size: usize = resource_info.base64size.parse::<usize>()?;
+
+        let recv_string: String =
+            self.kbs_trans_and_recv(trans_data, buffer_size, "Get Resource")?;
+
+        let data_bytes = base64::decode(recv_string)?;
+
+        if let Result::Ok(data_str) = std::str::from_utf8(&data_bytes) {
+            if let Result::Ok(err_info) = serde_json::from_str::<GetResourceErrorInfo>(data_str) {
+                return Err(anyhow!(
+                    "{}: The resource data is null, error reason: {}.",
+                    EAA_KBS_NAME,
+                    err_info.error
+                ));
+            }
+        }
+
+        Ok(data_bytes)
+    }
+
+    fn kbs_get_resource_info(&mut self, resource_name: &str) -> Result<ResourceInfo> {
+        let request = GetResourceInfoReq::new(resource_name);
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+
+        let recv_string: String =
+            self.kbs_trans_and_recv(trans_data, DEFAULT_RECV_BYTES_SIZE, "Get Resource Info")?;
+
+        let response = serde_json::from_str::<GetResourceInfoResponse>(recv_string.as_str())?;
+
+        match response.status.as_str() {
+            "OK" => {
+                if let Some(info) = response.data {
+                    Ok(info)
+                } else {
+                    return Err(anyhow!("{}: Resource info payload is null", EAA_KBS_NAME));
+                }
+            }
+            "Fail" => {
+                return Err(anyhow!(format!(
+                    "{}: {}",
+                    EAA_KBS_NAME,
+                    response.error.unwrap()
+                )))
+            }
+            _ => {
+                return Err(anyhow!(
+                    "Can not understand the GetResourceInfoResponse status"
+                ))
+            }
+        }
+    }
+
+    fn kbs_trans_and_recv(
+        &mut self,
+        trans_data: &[u8],
+        recv_size: usize,
+        error_info: &str,
+    ) -> Result<String> {
+        if trans_data.to_vec().is_empty() || recv_size == 0 {
+            return Err(anyhow!(
+                "Transmit data and recieve buffer can not be empty."
```

Comment:
Nit: Message is slightly inaccurate. Receive buffer can be empty, it just can't have length zero.